### ISSUE TITLE
Save best model with F1 score

### ISSUE THREAD

**Issue by fbernardini** [fuzailpalnak/building-footprint-segmentation#41]
Hi,
I'm using your scripts to train the model on a custom dataset.
Analyzing your code, i noticed that you use validation loss to save the best model. I would like to save it based on the F1 metric.
Could you please help me changing the script to address my request?

Thank you,
Francesca

**Comment by fuzailpalnak** [comment]
The easiest and fastest hack would be to add  ` valid_loss = -valid_metric["f1"]`  [here](https://github.com/fuzailpalnak/building-footprint-segmentation/blob/1998968eb3d5343c9bfd7528fb33287970804cde/building_footprint_segmentation/trainer.py#L101). If you want to make it user configurable for all other metrics, then there are few sections where changes are needed.

**Comment by fbernardini** [comment]
Thank you!
